src/algorithm/boundingbox.rs

```diff
@@ -0,0 +1,209 @@
+use num::{Float};
+
+use types::{Bbox, Point, MultiPoint, LineString, MultiLineString, Polygon, MultiPolygon};
+
+/// Calculation of the bounding box of a geometry.
+
+pub trait BoundingBox<T: Float> {
+    /// Return a Bounding Box of a geometry
+    ///
+    /// ```
+    /// use geo::{Point, LineString, Coordinate};
+    /// use geo::algorithm::boundingbox::BoundingBox;
+    ///
+    /// let mut vec = Vec::new();
+    /// vec.push(Point::new(40.02f64, 116.34));
+    /// vec.push(Point::new(42.02f64, 116.34));
+    /// vec.push(Point::new(42.02f64, 118.34));
+    /// let linestring = LineString(vec);
+    /// let bbox = linestring.bbox().unwrap();
+    ///
+    /// println!("Bbox top left coordinates: {}, {}", bbox.xmin, bbox.ymax);
+    /// println!("Bbox bottom right coordinates: {}, {}", bbox.xmax, bbox.ymin);
+    /// ```
+    ///
+    fn bbox(&self) -> Option<Bbox<T>>;
+}
+
+fn get_bbox<T>(vect: &Vec<Point<T>>) -> Option<Bbox<T>>
+    where T: Float
+{
+    if vect.is_empty() {
+        return None;
+    }
+    if vect.len() == 1 {
+        return Some(Bbox{xmin: vect[0].x(), ymax: vect[0].y(),
+                         xmax: vect[0].x(), ymin: vect[0].y()})
+    } else {
+        let (mut xmax, mut xmin) = (T::neg_infinity(), T::infinity());
+        let (mut ymax, mut ymin) = (T::neg_infinity(), T::infinity());
+        for pnt in vect.iter() {
+            let (px, py) = (pnt.x(), pnt.y());
+            if px > xmax {
+                xmax = px;
+            } else if px < xmin {
+                xmin = px;
+            }
+            if py > ymax {
+                ymax = py;
+            } else if py < ymin {
+                ymin = py;
+            }
```

Comment:
```rust
let mut x_min_max = (vect[0].x(), vect[0].x());
let mut y_min_max = (vect[0].y(), vect[0].y());
[...]
x_min_max = get_min_max(px, xmin, xmax);
y_min_max = get_min_max(py, ymin, ymax);
[...]
Some(Bbox{xmin: x_min_max.0, ymax: y_min_max.1, xmax: x_min_max.1, ymin: y_min_max.0})
```

fold version:
```rust
let mut iter = vect.into_iter();
let (init_x, init_y) = if let Some(p) = iter.next() {
    Bbox::new(p)
else {
    return None
};
iter.fold(init, |accu, p| accu.add(p))
```

Supposing there is a `Bbox::new(p: Point)` that create the bbox of a point and `Bbox::add(&self, p: Point) -> Self` that returns the Bbox containing self and p. Maybe you'll find a better name for `add`, I don't like it.